### ISSUE TITLE
Problem with runGist on mavericks

### ISSUE THREAD

**Issue by psychemedia** [rstudio/shiny#474]
Trying to use runGist on Mavericks, I get:

```
runGist('3239667')
Downloading https://gist.github.com/3239667/download
Error in runUrl(gistUrl, filetype = ".tar.gz", subdir = NULL, port = port,  : 
  Failed to download URL https://gist.github.com/3239667/download
In addition: Warning message:
In download.file(url, method = method, ...) :
  download had nonzero exit status
```

Trying the URL in a browser - https://gist.github.com/3239667/download - downloads fine? So the download is being blocked by something in the runGist() call?

**Comment by mossmatters** [comment]
I am also having this issue on Mac OS 10.10.4. I just tried updating to the newest version of R and updating shiny.

```R
library(shiny)
runGist('3239667')
```

```
Downloading https://gist.github.com/3239667/download
Error in runUrl(gistUrl, filetype = ".tar.gz", destdir = destdir, ...) : 
  Failed to download URL https://gist.github.com/3239667/download
In addition: Warning message:
In download.file(url, method = method, ...) :
  download had nonzero exit status
```

```R
sessionInfo()
```

```
R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.4 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_0.12.1

loaded via a namespace (and not attached):
[1] R6_2.1.0        htmltools_0.2.6 tools_3.2.1     Rcpp_0.11.6     digest_0.6.8    xtable_1.7-4    httpuv_1.3.2    mime_0.3
```

The curl -L command worked as above, and I can find the gist in my browser just fine.

**Comment by jcheng5** [comment]
@mossmatters If you upgrade using `devtools::install_github("rstudio/shiny")` it will start working again. Sorry, GitHub changed the behavior of their gist download endpoints, they're serving zips instead of tarballs now.

**Comment by kdaily** [comment]
I think this bug exists in `shiny::runGithub` as well (doesn't find a `tar.gz`, looks like Github serves up `zip` as well).

**Comment by pranasblk** [comment]
+1 To fix it instead of keeping closed.

On can try running fragments of code from their documentations http://shiny.rstudio.com/articles/persistent-data-storage.html

```
> shiny::runGist("c4db11d81f3c46a7c4a5")
Downloading https://gist.github.com/c4db11d81f3c46a7c4a5/download
wget: invalid option -- 'f'
Usage: wget [OPTION]... [URL]...

Try `wget --help' for more options.
Error in runUrl(gistUrl, filetype = ".zip", destdir = destdir, ...) : 
  Failed to download URL https://gist.github.com/c4db11d81f3c46a7c4a5/download
In addition: Warning message:
In utils::download.file(url, method = method, ...) :
  download had nonzero exit status
```
```
> library(shiny); sessionInfo()
R version 3.2.2 (2015-08-14)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_0.12.2

loaded via a namespace (and not attached):
[1] R6_2.1.1        htmltools_0.2.6 tools_3.2.2     Rcpp_0.12.1     digest_0.6.8    xtable_1.7-4    httpuv_1.3.3   
[8] mime_0.4       
```

**Comment by wch** [comment]
@kdaily, please see #996. If you think you're experiencing a different problem from that, please file a new issue.

**Comment by kdaily** [comment]
Think it's the same @wch!
